go-controller/pkg/node/gateway_localnet.go

```diff
@@ -156,27 +156,43 @@ func initLocalnetGateway(nodeName string, subnet *net.IPNet, wf *factory.WatchFa
 	if err != nil {
 		return err
 	}
-
-	var gatewayIP, gatewayNextHop net.IP
-	var gatewaySubnetMask net.IPMask
-	if utilnet.IsIPv6CIDR(subnet) {
-		gatewayIP = net.ParseIP(v6localnetGatewayIP)
-		gatewayNextHop = net.ParseIP(v6localnetGatewayNextHop)
-		gatewaySubnetMask = net.CIDRMask(v6localnetGatewaySubnetPrefix, 128)
-	} else {
-		gatewayIP = net.ParseIP(v4localnetGatewayIP)
-		gatewayNextHop = net.ParseIP(v4localnetGatewayNextHop)
-		gatewaySubnetMask = net.CIDRMask(v4localnetGatewaySubnetPrefix, 32)
+	// Flush any addresses on localnetGatewayNextHopPort first
+	if err := util.LinkAddrFlush(link); err != nil {
+		return err
 	}
-	gatewayIPCIDR := &net.IPNet{IP: gatewayIP, Mask: gatewaySubnetMask}
-	gatewayNextHopCIDR := &net.IPNet{IP: gatewayNextHop, Mask: gatewaySubnetMask}
 
-	// Flush any addresses on localnetBridgeNextHopPort and add the new IP address.
-	if err = util.LinkAddrFlush(link); err == nil {
-		err = util.LinkAddrAdd(link, gatewayNextHopCIDR)
-	}
-	if err != nil {
-		return err
+	var gatewayIPs []*net.IPNet
+	var nextHops []net.IP
+	for _, hostSubnet := range hostSubnets {
+		var gatewayIP, gatewayNextHop net.IP
+		var gatewaySubnetMask net.IPMask
+		if utilnet.IsIPv6CIDR(hostSubnet) {
+			gatewayIP = net.ParseIP(v6localnetGatewayIP)
+			gatewayNextHop = net.ParseIP(v6localnetGatewayNextHop)
+			gatewaySubnetMask = net.CIDRMask(v6localnetGatewaySubnetPrefix, 128)
+		} else {
+			gatewayIP = net.ParseIP(v4localnetGatewayIP)
+			gatewayNextHop = net.ParseIP(v4localnetGatewayNextHop)
+			gatewaySubnetMask = net.CIDRMask(v4localnetGatewaySubnetPrefix, 32)
+		}
+		gatewayIPCIDR := &net.IPNet{IP: gatewayIP, Mask: gatewaySubnetMask}
+		gatewayNextHopCIDR := &net.IPNet{IP: gatewayNextHop, Mask: gatewaySubnetMask}
+		gatewayIPs = append(gatewayIPs, gatewayIPCIDR)
```

Comment:
Does it make sense to use `[]` subscript operator as against append since we know the length of the `gatewayIPs` and `nextHops` slice apriori? `append` is incrementally slower than `[]`.

Reply:
it doesn't really matter; this only runs once